actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -59,7 +59,7 @@ def check_path_parameters!
       path_parameters.each do |key, value|
         next unless value.respond_to?(:valid_encoding?)
         unless value.valid_encoding?
-          raise ActionController::BadRequest, "Invalid parameter: #{key} => #{value}"
```

Comment:
Is not this the same thing that is being done in the new code but just with a different class? Why the different class is needed?